src/parallel/task_scheduler.cpp

```diff
@@ -78,22 +83,39 @@ struct ConcurrentQueue {
 
 void ConcurrentQueue::Enqueue(ProducerToken &token, shared_ptr<Task> task) {
 	lock_guard<mutex> lock(qlock);
-	q.push(std::move(task));
+	QueueProducerToken *token_addr = token.token.get();
```

Comment:
Can we use references instead of pointers here?

Reply:
Boaz just mentioned that a `forward_list` with tuples should probably be quicker here, as the amount of tasks in the queue should not be to large. Would you be OK with that as well?

Reply:
I don't really see a reason to switch from the `queue` given that we are only pulling tasks from one side.